### DISCUSSION TITLE
Mounting data

### DISCUSSION THREAD

**Discussion by ju5t** [spatie/laravel-livewire-wizard#6] [Ideas]
@freekmurze before I make a PR, here's my use case.

We're going to build a wizard that's based on an order. The wizard will need access to both the order and some of its relations. In a normal Livewire component this would look something like this:

```php
public function mount(Order $order)
{
    $this->order = $order;
    $this->order->load('items');
}
```

This isn't possible right now. So here's a rough proposal.

```php
public function mount(string $showStep = '')
{
    $step = empty($showStep)
            ? $this->stepNames()->first()
            : $showStep;
    
    $this->showStep($step, $this->getInitialState());
}

public function getInitialState(): array
{
    return [];
}
```

Is this something you would consider? It would allow you to implement `getInitialState()` and I think once `mount()` runs, you will have access to properties that have been set on the component.

Does this make any sense?

**Comment by freekmurze** [comment]
Thanks for your suggestion, do send a PR for this! 👍

**Comment by ju5t** [comment]
For future readers see #7.
